packages/firestore/src/api/database.ts

```diff
@@ -1905,23 +1906,25 @@ function throwDocChangesMethodError(): never {
   );
 }
 
-/**
- * This is technically overwriting the `Function.prototype.length` property of
- * `docChanges`. On IE11, the property is improperly defined with
- * `{ configurable: false }` which causes this line to throw. Wrap in a
- * try-catch to ensure that we still have a functional SDK.
- */
-try {
-  Object.defineProperty(QuerySnapshot.prototype.docChanges, 'length', {
-    get: () => throwDocChangesMethodError()
-  });
-} catch (err) {} // Ignore this failure intentionally
-
-if (typeof Symbol !== 'undefined') {
-  Object.defineProperty(QuerySnapshot.prototype.docChanges, Symbol.iterator, {
-    get: () => throwDocChangesMethodError()
-  });
-}
+const docChangesPropertiesToOverride = [
+  'length',
+  'forEach',
+  'map',
+  ...(typeof Symbol !== 'undefined' ? [Symbol.iterator] : [])
+];
+docChangesPropertiesToOverride.forEach(property => {
+  /**
+   * This will technically overwrite the `Function.prototype.length` property of
```

Comment:
Can you update the comment to indicate that "this" indicates "overwriting length".

Reply:
Thanks, I rewrote the comment to hopefully be clearer:

```
  /**
   * We are (re-)defining properties on QuerySnapshot.prototype.docChanges which
   * is a Function. This could fail, in particular in the case of 'length' which
   * already exists on Function.prototype and on IE11 is improperly defined with
   * `{ configurable: false }`. So we wrap this in a try/catch to ensure that we
   * still have a functional SDK.
   */
```

I'll hold off on trying to submit this until you make the publishToNpm() style thing go away.

Reply:
Thanks. I like it a lot!